### PR TITLE
avm2: Add support for SWZ files

### PR DESCRIPTION
**Pull request by EmperorBale**
This allows the Loader class to load SWZ files.

### CODE REVIEW

**Comment by hatal175** [comment]
Shouldn't this verify the digest in some form?

**Comment by EmperorBale** [comment]
> Shouldn't this verify the digest in some form?

Well, currently the digest field is just stubbed. From my understanding, that digest is just for caching purposes, and Ruffle currently does not do any caching, so there isn't really any point for Ruffle verifying it.

**Review by Aaron1011 — APPROVED** [review]

File: swf/src/read.rs [hunk 1]
```diff
@@ -28,6 +29,28 @@ pub fn parse_swf(swf_buf: &SwfBuf) -> Result<Swf<'_>> {
     })
 }
 
+/// Extracts an SWF inside of an SWZ file.
+pub fn extract_swz(input: &[u8]) -> Result<Vec<u8>> {
```
Comment (relrelb) [inline swf/src/read.rs:33]: Any reason not introducing this function directly in `loader.rs`, since it's used only there?

Comment (EmperorBale) [reply swf/src/read.rs:33]: Well I felt like SWZ was really just a wrapper for an SWF file, so it should be included as part of the SWF library. I can make it separate, but it feels weird to me

**Pull request merged**